flybywiresim/a32nx
# feat: automatic brightness and lighting

Fixes #[issue_no]

## Summary of Changes

- The automatic brightness functions of the EFB and ISIS are improved,
- The ISIS is refactored and a couple of bugs fixed (+/- keys no longer affect bugs when not on bugs screen)
- The lighting init code from the old EICAS is moved somewhere it will still run
- Automatic brightness level for cockpit lighting and displays on spawn (lights off/screens bright during the day, lights on and screens dimmer during the night, depending on ambient light level).

## Screenshots (if necessary)


Daytime spawn:
![20230222203758_1](https://user-images.githubusercontent.com/9995998/220555339-19deb0cc-d321-4b0b-a8be-82116b4327c7.jpg)
Midnight spawn:
![20230222210506_1](https://user-images.githubusercontent.com/9995998/220560052-87207a97-0e60-42a0-b289-faa19c214409.jpg)

## References






## Additional context



Discord username (if different from GitHub):

## Testing instructions

Ensure no preset auto-loading is enabled. 
- Spawn in different conditions and check that lighting and display brightness is always reasonable at spawn. 
- Ensure ISIS brightness controls and bugs menu still work okay.
- Setup a lighting preset in the flyPad and turn on auto-loading for it, ensure it auto-loads okay


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
